astropy/table/tests/test_index.py

```diff
@@ -4,18 +4,30 @@
 import numpy as np
 
 from .test_table import SetupData
-from ..bst import BST, FastRBT
+from ..bst import BST, FastRBT, FastBST
 from ..sorted_array import SortedArray
 from ..table import QTable, Row
 from ... import units as u
 from ...time import Time
 from ..column import BaseColumn
 from ...extern.six.moves import range
 
-
-@pytest.fixture(params=[BST, FastRBT, SortedArray])
-def engine(request):
-    return request.param
+try:
+    import bintrees
+except ImportError:
+    _HAS_BINTREES = False
+else:
+    _HAS_BINTREES = True
+
+
+if _HAS_BINTREES:
```

Comment:
Not sure if there is some pytest magic happening, but could you just define `engines` as a list based on `HAS_BINTREES` and then have `@pytest.fixture(params=engines)`.  I.e. don't repeat the whole function definition twice.  Just wondering.